### PR TITLE
poco: use openssl 1.1.1g instead of 1.0.2t

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **poco/ALL**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@madebr It fails to find libcrypto:

```
LINK : fatal error LNK1104: cannot open file 'libcrypto.lib' [C:\J\c3ipr\pr_1536_3_0_8\.conan\data\poco\1.10.1\_\_\build\6857d7ec15f5b00697dd620b27012b54602f9851\build_subfolder\Crypto\Crypto.vcxproj]
```

Did you check if Poco project is running with OpenSSL 1.1.1? It would be great!

**Comment by madebr** [comment]
@uilianries 
It worked on Linux (on the most recent poco release).
I'll have to check on Windows how to fix the problem.

**Comment by uilianries** [comment]
@madebr that's why I love Linux, everything builds there at first try.

File: recipes/poco/all/CMakeLists.txt [hunk 1]
```diff
@@ -5,4 +5,7 @@ message(STATUS "Conan CMake Wrapper")
 include("${CMAKE_CURRENT_SOURCE_DIR}/../conanbuildinfo.cmake")
 CONAN_BASIC_SETUP()
 
+# Disable automatic linking on MSVC
+add_definitions(-DPOCO_NO_AUTOMATIC_LIBS)
```
Comment (SSE4) [inline recipes/poco/all/CMakeLists.txt:9]: I think this should be part of `package_info` (`cpp_info.defines`), WDYT?

Comment (uilianries) [reply recipes/poco/all/CMakeLists.txt:9]: I think it's not required. POCO_NO_AUTOMATIC_LIBS is only used for #pragma comment(lib, ...). As we can provide the correct library order in cpp_info.libs, we don't need to export it.

Comment (madebr) [reply recipes/poco/all/CMakeLists.txt:9]: I think @SSE4 was right.
grep'ing for `POCO_NO_AUTOMATIC_LIBS` in the `include` folder of my package, turns up some results.

It is mostly used for linking to its own libraries,
but `include/Poco/Crypto/Crypto.h` contains a references to the ssl libraries.

**Pull request merged**